### PR TITLE
instructions for firedrake install on Mac

### PR DESCRIPTION
**Pull request by Ig-dolci**
---
name: "Bug fix or new feature"
description: ''
title: ''
labels: ''
assignees: ''

---

# Description
Please include a summary of the changes introduced by this PR.
Additionally include any relevant motivation and context. List any
dependencies that are required for this change, other PRs upon
which this may depend, and associated issues.

## Associated Pull Requests:
- Make a list (with links!)
- [Like this PR#691](https://github.com/OP2/PyOP2/pull/691)

## Fixes Issues:
- Make a list of issues (with links!)
- [Like this Issue#2824](https://github.com/firedrakeproject/firedrake/issues/2824)

If issues are fixed by this PR, prepend each of them with the word
"fixes", so they are automatically closed when this PR is merged. For
example "fixes #123, fixes #456".

# Checklist for author:

- [ ] I have linted the codebase (`make lint` in the `firedrake` source directory).
- [ ] My changes generate no new warnings.
- [ ] All of my functions and classes have appropriate docstrings.
- [ ] I have commented my code where its purpose may be unclear.
- [ ] I have included and updated any relevant documentation.
- [ ] Documentation builds locally (`make linkcheck; make html; make latexpdf` in `firedrake/docs` directory)
- [ ] I have added tests specific to the issues fixed in this PR.
- [ ] I have added tests that exercise the new functionality I have introduced
- [ ] Tests pass locally (`pytest tests` in the `firedrake` source directory) (useful, but not essential if you don't have suitable hardware).
- [ ] I have performed a self-review of my own code using the below guidelines.

# Checklist for reviewer:

- [ ] Docstrings present
- [ ] New tests present
- [ ] Code correctly commented
- [ ] No bad "code smells"
- [ ] No issues in parallel
- [ ] No CI issues (excessive parallelism/memory usage/time/warnings generated)
- [ ] Upstream/dependent branches and PRs are ready

Feel free to add reviewers if you know there is someone who is already aware of this work.

Please open this PR initially as a draft and mark as ready for review once CI tests are passing.

### CODE REVIEW

**Comment by Ig-dolci** [comment]
Reviewing the [instructions for firedrake installation on Mac](https://github.com/firedrakeproject/firedrake/wiki/Installation-instructions-for-macOS) is also necessary.

File: docs/source/download.rst [hunk 1]
```diff
@@ -110,7 +110,10 @@ have yet to create binary wheels for 3.11.x, but we have generated these
 for the major supported platforms.
 The installation script is tested on Ubuntu and MacOS X. On Ubuntu 22.04
 or later, the system installed Python 3 is supported and tested. On
-MacOS, the homebrew_ installed Python 3 is supported::
+MacOS, due to current issues with Xcode 15 and MPI_, we recommend to follow the
+instructions available 
+`here < https://github.com/firedrakeproject/firedrake/wiki/Installation-instructions-for-macOS>`_.
```
Comment (connorjward) [inline docs/source/download.rst:115]: I think that this should be put in a text box
```rst
.. note::
  Due to current issues with Xcode 15...
```

**Review by connorjward — CHANGES_REQUESTED** [review]
Before we update the website I think it would be good for someone else to follow the steps and make sure it solves the issue for them.

**Comment by JDBetteridge** [comment]
I have found additional issues with the installer that prevent the documented changes from working properly.
Essentially setting `LDFLAGS` in `PETSC_CONFIGURE_OPTIONS` may or may not (due to the unordered nature of Python `set()`) take precedence the `LDFLAGS` used to control BLAS linking in PETSc.

This requires a bigger fix, although probably needs discussion.

**Comment by Ig-dolci** [comment]
> I have found additional issues with the installer that prevent the documented changes from working properly. Essentially setting `LDFLAGS` in `PETSC_CONFIGURE_OPTIONS` may or may not (due to the unordered nature of Python `set()`) take precedence the `LDFLAGS` used to control BLAS linking in PETSc.
> 
> This requires a bigger fix, although probably needs discussion.

Hm. I see. So, downgrading CLT is the best solution for now since setting `LDFLAGS` in `PETSC_CONFIGURE_OPTIONS` is required only for CLT 15.

**Comment by JDBetteridge** [comment]
Anecdotally CLT 14.3.1 will not run on MacOS Sonoma, I can find forum posts about this but not an official statement* from Mac. This means users may need to downgrade their OS to fix in this manner.

*I also haven't looked very hard

**Comment by Ig-dolci** [comment]
> Anecdotally CLT 14.3.1 will not run on MacOS Sonoma, I can find forum posts about this but not an official statement* from Mac. This means users may need to downgrade their OS to fix in this manner.
> 
> *I also haven't looked very hard

Yep. The Xcode14 app is not working for Sonoma anymore, but we can still use clang14-0. I am with Sonoma and using the clang14 version here. That is a temporary fix until Xcode sorts this issue or, as you pointed out, we can try a bigger fix.

**Pull request closed without merging**